tpu-client/src/nonblocking/tpu_client.rs

```diff
@@ -1006,6 +1010,7 @@ async fn maybe_fetch_cache_info(
         maybe_cluster_nodes,
         maybe_epoch_schedule,
         maybe_slot_leaders,
+        first_slot: estimated_current_slot,
```

Comment:
cache the starting slot value that was used to pull down the slot leaders. This slot needs to map to the first entry in the `maybe_slot_leaders` vec